### PR TITLE
Use rct_widgetindex type

### PR DESCRIPTION
**Pull request by marijnvdwerf**

### CODE REVIEW

**Comment by marijnvdwerf** [comment]
Dunno. Found the type while working on the magic numbers, thought it would
be nice to have things be consistent.

Op ma 1 mei 2017 om 19:50 schreef Ted John <notifications@github.com>:

**Comment by janisozaur** [comment]
It very well may be. When we move this to c++, it can become `enum class`.

File: src/openrct2/input.c [hunk 1]
```diff
@@ -1049,7 +1049,7 @@ void process_mouse_over(sint32 x, sint32 y)
 	if (window != NULL) {
 		sint32 ebx, edi;
 		rct_window* subWindow;
-		sint32 widgetId = window_find_widget_from_point(window, x, y);
+        rct_widgetindex widgetId = window_find_widget_from_point(window, x, y);
```
Comment (Gymnasiast) [inline src/openrct2/input.c]: Botched alignment here.

**Comment by IntelOrca** [comment]
@janisozaur its an index and very often has arithmetic applied to it. `enum class` is no good for that. It can't even play nicely with bit masks.

**Comment by Gymnasiast** [comment]
@IntelOrca @janisozaur Apart from the need for a rebase, is this PR good to merge?

**Pull request merged**